test/extensions/filters/http/geoip/config_test.cc

```diff
@@ -0,0 +1,218 @@
+#include "envoy/extensions/filters/http/geoip/v3/geoip.pb.h"
+#include "envoy/extensions/filters/http/geoip/v3/geoip.pb.validate.h"
+
+#include "source/extensions/filters/http/geoip/config.h"
+#include "source/extensions/filters/http/geoip/geoip_filter.h"
+#include "source/extensions/filters/http/geoip/geoip_provider_config.h"
+
+#include "test/extensions/filters/http/geoip/mocks.h"
+#include "test/extensions/filters/http/geoip/utils.h"
+#include "test/mocks/server/factory_context.h"
+#include "test/test_common/registry.h"
+#include "test/test_common/test_runtime.h"
+
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+
+using ::testing::AllOf;
+using ::testing::Truly;
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace Geoip {
+namespace {
+
+using GeoipFilterConfig = envoy::extensions::filters::http::geoip::v3::Geoip;
+
+MATCHER_P(HasUseXff, expected, "") {
+  auto filter = std::static_pointer_cast<GeoipFilter>(arg);
+  if (GeoipFilterPeer::useXff(*filter) == expected) {
+    return true;
+  }
+  *result_listener << "expected useXff=" << expected << " but was "
+                   << GeoipFilterPeer::useXff(*filter);
+  return false;
+}
+
+MATCHER_P(HasXffNumTrustedHops, expected, "") {
+  auto filter = std::static_pointer_cast<GeoipFilter>(arg);
+  if (GeoipFilterPeer::xffNumTrustedHops(*filter) == static_cast<uint32_t>(expected)) {
+    return true;
+  }
+  *result_listener << "expected useXff=" << expected << " but was "
+                   << GeoipFilterPeer::useXff(*filter);
+  return false;
+}
+
+MATCHER_P(HasGeoHeader, expected_header, "") {
+  auto filter = std::static_pointer_cast<GeoipFilter>(arg);
+  auto geo_headers = GeoipFilterPeer::geoHeaders(*filter);
+  bool header_found = false;
+  for (auto header : geo_headers) {
+    if (header == expected_header) {
+      header_found = true;
```

Comment:
Can just return true here, not need header_found, and add some *result_listener output before return false after the loop.